cosmos-tx/src/crypto/compact_bit_array.rs

```diff
@@ -0,0 +1,40 @@
+//! Compact bit array.
+
+use cosmos_sdk_proto::cosmos;
+
+/// [`CompactBitArray`] is an implementation of a space efficient bit array.
+///
+/// This is used to ensure that the encoded data takes up a minimal amount of
+/// space after proto encoding.
+#[derive(Clone, Debug, PartialEq)]
+pub struct CompactBitArray {
+    // TODO(tarcieri): better internal representation for this, e.g. `bitvec`
```

Comment:
Thought I'd postpone making this decision for now, although I'd probably recommend `bitvec`.